arkouda/pdarraycreation.py

```diff
@@ -543,8 +544,12 @@ def full(
     >>> ak.full(5, 5, dtype=ak.bool)
     array([True, True, True, True, True])
     """
+
     if not np.isscalar(size):
         raise TypeError(f"size must be a scalar, not {size.__class__.__name__}")
+    if type(fill_value) == str:
```

Comment:
it doesn't make a difference in this case but in general we try to use `isinstance` for checks like this. For this case that'd look like:
```python
if isinstance(fill_value, str):
```

There's a couple reasons for this, it makes it easier to check multiple types at once. So for instance if we wanted to act if `fill_value` is any of `str`, `int`, or `bool`, then 
```python
# checking the type directly is much longer
if type(fill_value) == str or type(fill_value) == int or type(fill_value) == bool:
# than an isinstance check
if isinstance(fill_value, (str, int, bool)):
```
It also correctly handles subclasses (classes that inherit from another) better, for example
```python
>>> class rectangle:
     def __init__(self, length, width):
         self.length = length
         self.width = width

>>> class square(rectangle):
     # rectangle where both sides are same size
     def __init__(self, side_len):
         rectangle.__init__(self, side_len, side_len)
         self.side_len = side_len

# isinstance knows a square is a rectangle and a rectangle is not necessarily a square
>>> isinstance(square(10), rectangle)
True
>>> isinstance(rectangle(5, 10), square)
False

# but checking the type directly does not
>>> type(square(10)) == rectangle
False
```


A good example of this in arkouda is `DateTime` which inherits from `pdarray`